core/src/main/java/com/sequenceiq/distrox/v1/distrox/service/upgrade/DistroXUpgradeAvailabilityService.java

```diff
@@ -150,7 +151,8 @@ private List<ImageInfoV4Response> validateRangerRazCandidates(String accountId,
 
     private List<ImageInfoV4Response> filterCandidates(String accountId, Stack stack, UpgradeV4Request request, UpgradeV4Response upgradeV4Response) {
         BlueprintUpgradeOption upgradeOption = stack.getCluster().getBlueprint().getBlueprintUpgradeOption();
-        if (upgradeOption != BlueprintUpgradeOption.GA) {
+        ResourceStatus resourceStatus = stack.getCluster().getBlueprint().getStatus();
+        if (ResourceStatus.USER_MANAGED != resourceStatus && upgradeOption != BlueprintUpgradeOption.GA) {
             LOGGER.debug("Running filtering logic as upgrade option is {}, not GA", upgradeOption);
```

Comment:
nit: it would be nice to update the log line as well with something like "..., not GA and the blueprint is not a custom one".